src/components/ListboxGrid/distribute-resources.ts

```diff
@@ -1,61 +1,232 @@
+  const canExpand = size.height > EXPANDED_HEIGHT;
+  const maxColumns = getMaxColumns(size);
+  const maxPerColumn = getMaxItemsPerColumn(size);
+  const usedCount = getColumnItemsCount(columns);
+
+  if (canExpand && doesAllFit(maxPerColumn, maxColumns - columns.length - 1, size.dimensionCount - usedCount - 1)) {
+    columns.push({
+      expand: true,
+      itemCount: 1,
+    });
+    if (usedCount + 1 < size.dimensionCount) {
+      columns = calculateColumns(size, columns, zoomEnabled);
+    }
   } else {
-    const sliceIndex = 1 - columnCount;
-    firstColumn = resources.slice(0, sliceIndex);
-    restOfColumns = resources.slice(sliceIndex);
-  }
-  return { firstColumn, restOfColumns };
-};
-
-// Remove folded items that don't fit in height
-const removeOverflowingItems = (height: number, foldedItemHeight: number, count: number, rescources: IListboxResource[]) => {
-  const sliceIndex = (height / foldedItemHeight) - count - 2;
-  return rescources.slice(0, sliceIndex);
-};
-
-export const distributeResourcesInFirstCoulumn = (rescources: IListboxResource[], height: number, foldedCount: number, cExpanded: IListboxResource[], cFolded: IListboxResource[]) => {
-  let expanded: IListboxResource[] = [...cExpanded];
-  let folded: IListboxResource[] = [...cFolded];
-  let expandButton = false;
-  const count = rescources.length;
-  const foldedItemHeight = 60;
-  const foldedItemsHeight = () => (foldedCount ? (foldedItemHeight * foldedCount) + foldedItemHeight : 0);
-  const expandedCount = count - foldedCount;
-  const hysteresis = () => (count === 1 ? 30 : 110);
-  const expandedFactor = () => expandedCount * 30;
-  const padOneFolded = () => (foldedCount === 1 ? 20 : 0);
-  const foldLimit = foldedItemsHeight() + (expandedCount * 140) + expandedFactor();
-  const expandLimit = foldedItemsHeight() + (expandedCount * 140) + expandedFactor() + hysteresis() + padOneFolded();
-
-  if (height < foldLimit) {
-    if (height < foldedItemHeight * count) {
-      folded = removeOverflowingItems(height, foldedItemHeight, count, rescources);
-      expandButton = true;
-    } else {
-      folded = rescources.slice(0, foldedCount + 1);
-      expanded = rescources.slice(foldedCount + 1, rescources.length);
+    let itemCount;
+
+    // Default case with zoom enabled, now overflow and (...) to zoom object
+    itemCount = Math.min(size.dimensionCount - usedCount, maxPerColumn);
+
+    // TODO:
+    // if (ResponsiveState.isSmallDevice) {
+    // On small device all items is in a single column and overflow scrolled
+    // itemCount = size.dimensionCount - usedCount;
+    // } else if (!zoomEnabled) {
+    if (!zoomEnabled) {
+      // Zoom not enabled (mashup), items evenly distributed and overflow scrolled
+      // distribute the remaining items evenly
+      itemCount = Math.ceil((size.dimensionCount - usedCount) / (maxColumns - columns.length));
+      // to make sure all items in the columns are visible QB-2993
```

Comment:
Are these comments relevant still?
Any way we can avoid re-assigning `itemCount` by creating new, and more descriptive variables like:
```
const initialItemCount = …
const maxItemCount = …
const itemCount = …
```
Then we could get rid of the comments as well.